src/test/suite/oidc.test.ts

```diff
@@ -0,0 +1,431 @@
+  return createHash('sha256').update(input).digest('hex').slice(0, 12);
+}
+
+// Need to be provided via CI env because we can't get a hold for node.js exec
+// path in our tests - they run inside a vscode process
+const browserShellCommand = process.env.BROWSER_AUTH_COMMAND;
+
+const UNIQUE_TASK_ID =
+  process.env.GITHUB_RUN_ID && process.env.GITHUB_RUN_NUMBER
+    ? `${process.env.GITHUB_RUN_ID}-${process.env.GITHUB_RUN_NUMBER}`
+    : '';
+const defaultClusterOptions: MongoClusterOptions = {
+  topology: 'standalone',
+  tmpDir: path.join(os.tmpdir(), `vscode-tests-${hash(UNIQUE_TASK_ID)}-data`),
+  logDir: process.env.MONGODB_RUNNER_LOGDIR,
+  version: process.env.MONGODB_VERSION,
+};
+
+const DEFAULT_TOKEN_PAYLOAD = {
+  expires_in: 3600,
+  payload: {
+    // Define the user information stored inside the access tokens
+    groups: ['testgroup'],
+    sub: 'testuser',
+    aud: 'resource-server-audience-value',
+  },
+};
+
+suite('OIDC Tests', function () {
+  this.timeout(50000);
+
+  const extensionContextStub = new ExtensionContextStub();
+  const testStorageController = new StorageController(extensionContextStub);
+  const testTelemetryService = new TelemetryService(
+    testStorageController,
+    extensionContextStub
+  );
+  const testConnectionController = new ConnectionController({
+    statusView: new StatusView(extensionContextStub),
+    storageController: testStorageController,
+    telemetryService: testTelemetryService,
+  });
+  let showInformationMessageStub: SinonStub;
+  const sandbox = sinon.createSandbox();
+
+  // OIDC related variables
+  let getTokenPayload: typeof oidcMockProviderConfig.getTokenPayload = () =>
+    DEFAULT_TOKEN_PAYLOAD;
+  let overrideRequestHandler: typeof oidcMockProviderConfig.overrideRequestHandler;
+  let oidcMockProviderConfig: OIDCMockProviderConfig;
+  let oidcMockProvider: OIDCMockProvider;
+  let oidcMockProviderEndpointAccesses: Record<string, number>;
+
+  let i = 0;
+  let tmpdir: string;
+  let cluster: MongoCluster;
+  let connectionString: string;
+
+  let createTerminalStub: SinonStub;
+  let sendTextStub: SinonStub;
+
+  before(async function () {
+    if (process.platform !== 'linux') {
+      // OIDC is only supported on Linux in the 7.0+ enterprise server.
+      return this.skip();
+    }
+
+    oidcMockProviderEndpointAccesses = {};
+    oidcMockProviderConfig = {
+      getTokenPayload(metadata: Parameters<typeof getTokenPayload>[0]) {
+        return getTokenPayload(metadata);
+      },
+      overrideRequestHandler(url, req, res) {
+        const { pathname } = new URL(url);
+        oidcMockProviderEndpointAccesses[pathname] ??= 0;
+        oidcMockProviderEndpointAccesses[pathname]++;
+        return overrideRequestHandler?.(url, req, res);
+      },
+    };
+    oidcMockProvider = await OIDCMockProvider.create(oidcMockProviderConfig);
+
+    tmpdir = path.join(
+      os.tmpdir(),
+      `vscode-oidc-${Date.now().toString(32)}-${++i}`
```

Comment:
Since this is in the `before` function isn't `i` always going to be 1 then?

Reply:
Yea, thank you I will update it.